account-integrations/safe/test/hardhat/SafeBlsPlugin.test.ts

```diff
@@ -0,0 +1,197 @@
+import { ethers } from "hardhat";
+import { expect } from "chai";
+import { AddressZero } from "@ethersproject/constants";
+import { utils } from "ethers-v5";
+import { UserOperationStruct } from "@account-abstraction/contracts";
+import { calculateProxyAddress } from "../../utils/calculateProxyAddress";
+import { signer as hubbleBlsSigner } from '@thehubbleproject/bls';
+import { getUserOpHash } from "@account-abstraction/utils";
+
+import { SafeProxyFactory } from "../../typechain-types/lib/safe-contracts/contracts/proxies/SafeProxyFactory";
+import { Safe } from "../../typechain-types/lib/safe-contracts/contracts/Safe";
+import { EntryPoint } from "../../typechain-types/lib/account-abstraction/contracts/core/EntryPoint";
+
+const BLS_PRIVATE_KEY = '0xdbe3d601b1b25c42c50015a87855fdce00ea9b3a7e33c92d31c69aeb70708e08';
+const MNEMONIC = "test test test test test test test test test test test junk";
+
+let safeProxyFactory: SafeProxyFactory;
+let safe: Safe;
+let entryPoint: EntryPoint;
+
+describe("SafeBlsPlugin", () => {
+  const setupTests = async () => {
+    safeProxyFactory = await (
+        await ethers.getContractFactory("SafeProxyFactory")
+    ).deploy();
+    safe = await (
+        await ethers.getContractFactory("Safe")
+    ).deploy();
+    entryPoint = await (
+        await ethers.getContractFactory("EntryPoint")
+    ).deploy();
+
+    const provider = ethers.provider;
+    const userWallet = ethers.Wallet.fromPhrase(MNEMONIC).connect(
+      provider
+    );
+
+    return {
+        provider,
+        userWallet,
+        entryPoint,
+        safe,
+        safeProxyFactory,
```

Comment:
If we're creating a `safeProxyFactory` variable and others before the describe block and then assigning to them, is there a reason to return them in this function as well?